src/rv32_template.c

```diff
@@ -4,42 +4,73 @@
  */
 
 /* Internal */
-RVOP(nop, { rv->X[rv_reg_zero] = 0; })
+RVOP(nop, { rv->X[rv_reg_zero] = 0; }, {RVCODEGEN(nop, {})})
 
 /* LUI is used to build 32-bit constants and uses the U-type format. LUI
  * places the U-immediate value in the top 20 bits of the destination
  * register rd, filling in the lowest 12 bits with zeros. The 32-bit
  * result is sign-extended to 64 bits.
  */
-RVOP(lui, { rv->X[ir->rd] = ir->imm; })
+RVOP(lui, { rv->X[ir->rd] = ir->imm; }, {RVCODEGEN(lui, {
```

Comment:
Instead of `RVCODEGEN`, use `X64` identifier instead. Later, we may integrate A64 backend as well.

Reply:
Can you avoid the repeated `lui` declaration?

Reply:
> Can you avoid the repeated `lui` declaration?

Does you means rewrite to `RVCODEGEN({})`?

Reply:
> Instead of `RVCODEGEN`, use `X64` identifier instead. Later, we may integrate A64 backend as well.

Actually, every lines of machine code are generated from API, so we can switch the target machine in the API.

Reply:
I expect the instruction such as "lui" appears once within the scope of `RVOP` instead of twice.